### PR TITLE
[ROCm] Update ROCm skip decorators

### PR DESCRIPTION
**Pull request by lcskrishna**
This PR adds a msg argument for skipIfRocm and skipCUDAIfRocm. 

cc: @jataylo @jithunnair-amd 

cc @jeffdaily @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @dllehr-amd @jataylo @hongxiayang

### CODE REVIEW

**Comment by jataylo** [comment]
Hey @lcskrishna looks great, but it seems that your @skipIfRocm example doesn't trigger the skip message because it hits another skip reason, lets find another test case for it.
```
@skipCUDAIfRocm
test_nn.py::TestNNDeviceTypeCUDA::test_ctc_loss_cudnn_cuda SKIPPED [0.0005s] (skipCUDAIfRocm: skipped Cudnn test on ROCm) [ 60%]

@skipIfRocm
test_cuda_multigpu.py::TestCudaMultiGPU::test_stream_event_nogil SKIPPED [0.0001s] (detected only one GPU) [ 54%]
```

**Comment by lcskrishna** [comment]
Here are the test results for skipIfRocm and skipCUDAIfRocm with msg tags enabled from CI logs. 

```
skipCUDAIfRocm
2023-08-03T17:48:04.3150162Z test_nn.py::TestNNDeviceTypeCUDA::test_ctc_loss_cudnn_cuda SKIPPED [0.0004s] (skipCUDAIfRocm: skipped Cudnn test on ROCm)

skipIfRocm
2023-08-03T20:06:59.4187523Z test_cuda_expandable_segments.py::TestCuda::test_streaming_backwards_multiple_streams <- test/test_cuda.py SKIPPED [0.0016s] (skipIfRocm: flakey on ROCm https://github.com/pytorch/pytorch/issues/53190) [ 75%]
```

**Comment by jataylo** [comment]
Just checked with the latest update and both the decorators are outputting correctly
```
test_nn.py::TestNNDeviceTypeCUDA::test_ctc_loss_cudnn_cuda SKIPPED [0.0004s] (skipCUDAIfRocm: skipped Cudnn test on ROCm) [ 60%]  
test_cuda_expandable_segments.py::TestCuda::test_streaming_backwards_multiple_streams <- test/test_cuda.py SKIPPED [0.0016s] (skipIfRocm: flakey on ROCm https://github.com/pytorch/pytorch/issues/53190) [ 75%]
```

**Review by jataylo — APPROVED** [review]

**Review by jeffdaily — APPROVED** [review]

**Review by pruthvistony — APPROVED** [review]

**Comment by pruthvistony** [comment]
@malfet ,
Can you please help on reviewing this PR.

File: torch/testing/_internal/common_device_type.py [hunk 1]
```diff
@@ -1339,8 +1339,13 @@ def skipCUDAIfNoMagmaAndNoLinalgsolver(fn):
         return skipCUDAIfNoMagma(fn)
 
 # Skips a test on CUDA when using ROCm.
-def skipCUDAIfRocm(fn):
-    return skipCUDAIf(TEST_WITH_ROCM, "test doesn't currently work on the ROCm stack")(fn)
+def skipCUDAIfRocm(func=None, msg="test doesn't currently work on the ROCm stack"):
```
Comment (albanD) [inline torch/testing/_internal/common_device_type.py:1342]: Make msg a kwarg only arg to make sure it is no interpreted as a fn.
```suggestion
def skipCUDAIfRocm(func=None, *, msg="test doesn't currently work on the ROCm stack"):
```

File: torch/testing/_internal/common_utils.py [hunk 2]
```diff
@@ -1265,14 +1265,20 @@ def has_corresponding_torch_dtype(np_dtype):
...
 
-def skipIfRocm(fn):
-    @wraps(fn)
-    def wrapper(*args, **kwargs):
-        if TEST_WITH_ROCM:
-            raise unittest.SkipTest("test doesn't currently work on the ROCm stack")
-        else:
-            fn(*args, **kwargs)
-    return wrapper
+def skipIfRocm(func=None, msg="test doesn't currently work on the ROCm stack"):
```
Comment (albanD) [inline torch/testing/_internal/common_utils.py:1268]: Same about kwarg only.

**Review by albanD — COMMENTED** [review]
SGTM
Small detail to improve the API.

File: torch/testing/_internal/common_device_type.py (hunk 1 above)
Comment (lcskrishna) [reply torch/testing/_internal/common_device_type.py:1342]: Thank you @albanD, this makes more sense. Will update the PR with this change.

**Comment by lcskrishna** [comment]
@albanD The PR is ready for review the change suggested before.

**Review by albanD — APPROVED** [review]
Looks good, thank!

**Pull request closed without merging**